tests/linalg_test.py

```diff
@@ -757,6 +757,34 @@ def testMatrixRank(self, shape, dtype, rng_factory):
     self._CompileAndCheck(np.linalg.matrix_rank, args_maker,
                           check_dtypes=False, rtol=1e-3)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_shapes={}".format(
+           ','.join(jtu.format_shape_dtype_string(s, dtype) for s in shapes)),
+       "shapes": shapes, "dtype": dtype, "rng_factory": rng_factory}
+      for shapes in [
+        [(3, ), (3, 1)],  # quick-out codepath
+        [(1, 3), (3, 5), (5, 2)],  # multi_dot_three codepath
+        [(1, 3), (3, 5), (5, 2), (2, 7), (7, )]  # dynamic programming codepath
+      ]
+      for dtype in float_types + complex_types
+      for rng_factory in [jtu.rand_default]))
+  def testMultiDot(self, shapes, dtype, rng_factory):
+    rng = rng_factory()
+    _skip_if_unsupported_type(dtype)
+    args_maker = lambda: [[rng(shape, dtype) for shape in shapes]]
+
+    tol = {onp.float32: 1e-4, onp.float64: 1e-10,
+           onp.complex64: 1e-4, onp.complex128: 1e-10}
+    if jtu.device_under_test() == "tpu":
+      tol[onp.float32] = tol[onp.complex64] = 1e-2
```

Comment:
As a general proposition, I prefer to set `precision=lax.Precision.HIGHEST` rather than lowering the TPU tolerance. This will require that you plumb a new `precision` option into these APIs.

Stephan has argued that it should be the default, and I see merit to that idea.

Reply:
+1 it would be great to add `precision` as a keyword only argument to `multi_dot` for consistency with `jnp.dot`, etc

Reply:
Done